### PR TITLE
ENH: Improvements to Topics Ratings

### LINKED ISSUE #398: Question: Should a user be able to change a topic rating?
## Please summarize your question in one sentence
Should a user be able to change the 1-to-5-star rating they have previously given to a topic?

## Give a more extended description
Currently, a user can "vote" only once with a rating on a topic.

Should they be allowed to change their rating?

### PR DESCRIPTION
**Pull request by johnhenley**
### Description of PR...
Several updates to how topic ratings are handled.

## Changes made
- add DAL2 entity/controller for activeforums_Topics_Ratings
- migrate all stored procedure-based CRUD code to use DAL2 controller
- update indexes on activeforums_Topics_Ratings
- remove database stored procedures
- migrate handler code to use web api
- Allow user to change their rating (previously once rated, never changeable)

## How did you test these updates?  
Adding topics and changing rates; reviewing backend database

## PR Template Checklist

- [ ] Fixes Bug
- [X] Feature solution
- [X] Other
- [ ] Requires documentation updates  
- [ ] I've updated the documentation already  

## Please mark which issue is solved

Close #398 
Addresses (in part) #53 #266 #267

### CODE REVIEW

**Comment by WillStrohl** [comment]
@johnhenley This one is marked as 8.0.3, so I'm going to wait on the PR review for the moment.

File: Dnn.CommunityForums/DnnCommunityForums.csproj [hunk 1]
```diff
@@ -1012,6 +1012,8 @@
     <None Include="sql\07.00.02.SqlDataProvider" />
     <None Include="sql\07.00.00.SqlDataProvider" />
     <None Include="sql\07.00.02.SqlDataProvider" />
+    <None Include="sql\08.00.03.SqlDataProvider" />
```
Comment (WillStrohl) [inline Dnn.CommunityForums/DnnCommunityForums.csproj:1015]: Should this version be here? Should we put that SQL into the 8.1 SQL?

**Review by WillStrohl — APPROVED** [review]
I only have the one question about the 8.0.3 SqlDataProvider file.

**Comment by WillStrohl** [comment]
@johnhenley I meant to submit this as review pending, but it's appearing as reviewed & approved now.  I still want us to address the 8.0.3 SqlDataProvider reference first, since we're trying to release 8.1.0.

File: Dnn.CommunityForums/DnnCommunityForums.csproj (hunk 1 above)
Comment (johnhenley) [reply Dnn.CommunityForums/DnnCommunityForums.csproj:1015]: > Should this version be here? Should we put that SQL into the 8.1 SQL?

Please merge 636, 637, and 638 first ....
image

**Comment by johnhenley** [comment]
> @johnhenley I meant to submit this as review pending, but it's appearing as reviewed & approved now. I still want us to address the 8.0.3 SqlDataProvider reference first, since we're trying to release 8.1.0.
Ah. Thanks. I misunderstood. I copied 08.00.03.SqlDataProvider to 08.01.00.SqlDataProvider and forgot to delete the 08.00.03 one. I just committed to remove it.

**Review by WillStrohl — APPROVED** [review]
Looks good!

**Pull request merged**